### PR TITLE
Pagination Feature

### PR DESCRIPTION
**Pull request by Maurice22**
Implements #18 

Also redesigns the way requests are filtered by filtering them server-side.

@kenodressel Might this be counterproductive in terms of readcount?

### CODE REVIEW

**Comment by Maurice22** [comment]
@kenodressel Now on the most recent version. I also added a "ScrollToTop"-Button component from npm that is visible on all sites as soon as you start scrolling down.

File: src/components/FilteredList.js [hunk 1]
```diff
@@ -7,62 +7,139 @@ import LocationInput from './LocationInput';
...
-  const [entries, setEntries] = useState([
-    {
-      id: 'placeholder-id',
-    }]);
-  const [filteredEntries, setFilteredEntries] = useState([
-    {
-      id: 'placeholder-id',
-    }]);
+  const [entries, setEntries] = useState([]);
+  const [scheduledSearch, scheduleSearch] = useState([]);
```
Comment (kenodressel) [inline src/components/FilteredList.js]: should be `setScheduledSearch` to follow conventions

File: src/components/FilteredList.js [hunk 2]
```diff
@@ -7,62 +7,139 @@ import LocationInput from './LocationInput';
...
+        queryResult = collection.orderBy('d.timestamp', 'desc');
+        console.error('Error', error);
+      }
+    } else {
+      queryResult = collection;
+
+      if (location && location !== '') {
+        queryResult = queryResult.orderBy('d.plz', 'asc');
+
+        var lowerLimit = Number.parseInt(location) * (Math.pow(10, Math.max(5 - location.length, 0)));
```
Comment (kenodressel) [inline src/components/FilteredList.js]: this looks to me like we can only search for postal codes with 5 digits, but we should be also able to find 4 digit ones

File: src/components/FilteredList.js [hunk 3]
```diff
@@ -7,62 +7,139 @@ import LocationInput from './LocationInput';
...
+      }
+      scheduleSearch(setTimeout(() => {
+        loadFilteredData(buildQuery(address));
+      }, 500));
     }
   };
 
   const NoHelpNeeded = (props) => {
-    return <div className="w-full text-center my-10">In {location} wird gerade keine Hilfe gebraucht!</div>
+    return <div className="w-full text-center my-10">In {location} wird gerade aktuell keine Hilfe gebraucht!</div>
```
Comment (kenodressel) [inline src/components/FilteredList.js]: lets also add `font-open-sans` here

**Comment by Maurice22** [comment]
@kenodressel  Changes incorporated

**Pull request merged**